src/Type/ArrayType.php

```diff
@@ -442,6 +442,10 @@ public function shuffleArray(): Type
 
 	public function sliceArray(Type $offsetType, Type $lengthType, TrinaryLogic $preserveKeys): Type
 	{
+		if ($preserveKeys->no() && $this->keyType->isInteger()->yes()) {
+			return TypeCombinator::intersect(new self(new IntegerType(), $this->itemType), new AccessoryArrayListType());
```

Comment:
Could be replaced with `$this->getValuesArray()` if it should be more DRY. I wasn't sure, but this felt a bit less "hidden" to me.